ipnsfs/dir.go

```diff
@@ -0,0 +1,343 @@
+package ipnsfs
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	dag "github.com/jbenet/go-ipfs/merkledag"
+	ft "github.com/jbenet/go-ipfs/unixfs"
+	ufspb "github.com/jbenet/go-ipfs/unixfs/pb"
+)
+
+var ErrNotYetImplemented = errors.New("not yet implemented")
+var ErrInvalidChild = errors.New("invalid child node")
+
+type Directory struct {
+	fs     *Filesystem
+	parent childCloser
+
+	childDirs map[string]*Directory
+	files     map[string]*File
+
+	lock sync.Mutex
+	node *dag.Node
+
+	name string
+}
+
+func NewDirectory(name string, node *dag.Node, parent childCloser, fs *Filesystem) *Directory {
+	return &Directory{
+		fs:        fs,
+		name:      name,
+		node:      node,
+		parent:    parent,
+		childDirs: make(map[string]*Directory),
+		files:     make(map[string]*File),
+	}
+}
+
+// Open opens a file at the given path 'tpath'
+func (d *Directory) Open(tpath []string, mode int) (*File, error) {
+	if len(tpath) == 0 {
+		return nil, ErrIsDirectory
+	}
+	if len(tpath) == 1 {
+		fi, err := d.childFile(tpath[0])
```

Comment:
And it will still be tried to create (will call `NewFile`). I suggest adding:

```go
if err == <error that indicates dir> {
  return nil, ErrIsDirectory
} 
```
here